pretyping/detyping.ml

```diff
@@ -887,7 +887,7 @@ and detype_r d flags avoid env sigma t =
           try
             let c = Retyping.expand_projection (snd env) sigma p c [] in
             DAst.get (detype d flags avoid env sigma c)
-          with Retyping.RetypeError _ -> noparams ()
+          with _ -> noparams ()
```

Comment:
Catch-all clauses for exceptions are very bad™. At the very least you should check for non-criticality.

Reply:
The exception in https://github.com/coq/coq/issues/16716 is critical though
https://github.com/coq/coq/blob/90caf092f1946a81d3a104b6744faeef84ea2c4d/engine/evd.ml#L926

Reply:
This this is a hack, and the fix should be more principled.